samples/drivers/video/capture/src/main.c

```diff
@@ -147,6 +166,31 @@ int main(void)
 	fmt.width = CONFIG_VIDEO_FRAME_WIDTH;
 #endif
 
+	/*
+	 * Check (if possible) if targeted size is same as crop
+	 * and if compose is necessary
+	 */
+	sel.target = VIDEO_SEL_TGT_CROP;
+	err = video_get_selection(video_dev, &sel);
+	if (err == 0) {
+		if (sel.rect.width != fmt.width || sel.rect.height != fmt.height) {
+			sel.target = VIDEO_SEL_TGT_COMPOSE;
+			sel.rect.left = 0;
+			sel.rect.top = 0;
+			sel.rect.width = fmt.width;
+			sel.rect.height = fmt.height;
+			err = video_set_selection(video_dev, &sel);
+			if (err != -ENOSYS && err < 0) {
+				LOG_ERR("Unable to set selection compose");
+				return 0;
+			}
+		}
+	} else if (err != -ENOSYS && err < 0) {
+		LOG_ERR("Unable to get selection crop");
+		return 0;
+	}
```

Comment:
[non-blocking]
Proposal for making the error-handling idioms appear, just a minor nit:
```c
err = video_get_selection();            //
if (err < 0 && err != -ENOSYS) {        // well-known C error handling idiom
    LOG_ERR("That did not work");       //
    return 0;                           //
}                                       //

if (err == 0) {
    err = video_set_selection();        //
    if (err < 0 && err != -ENOSYS) {    // well-known C error handling idiom
        LOG_ERR("That did not work");   //
        return 0;                       //
    }                                   //
}
```

Reply:
Sometimes what is logical to write is not the same as what is logical to read. I got myself tricked in this exactly earlier.